python/metatensor-operations/metatensor/operations/sort.py

```diff
@@ -0,0 +1,231 @@
+            axes_list = ["samples", "components", "properties"]
+        elif axes not in ["samples", "components", "properties"]:
+            raise ValueError(
+                f"input parameter 'axes' may only be one of the strings 'samples', "
+                f"components' or 'properties' but is '{axes}'"
+            )
+        else:
+            axes_list = [axes]
+    elif isinstance(axes, list):
+        for axis in axes:
+            if axis not in ["samples", "components", "properties"]:
+                raise ValueError(
+                    f"input parameter 'axes' may only contain the strings 'samples', "
+                    f"components' or 'properties' but contains '{axis}'"
+                )
+        axes_list = axes
+    else:
+        raise TypeError("input paramater 'axes' should be of type str or list of str")
+
+    result_block = _sort_single_block(block, axes_list, descending)
+
+    for parameter, gradient in block.gradients():
+        if len(gradient.gradients_list()) != 0:
+            raise NotImplementedError("gradients of gradients are not supported")
+
+        result_block.add_gradient(
+            parameter=parameter,
+            gradient=_sort_single_block(gradient, axes_list, descending),
+        )
+
+    return result_block
+
+
+def sort(
+    tensor: TensorMap,
+    axes: Union[str, List[str]] = "all",
+    descending: bool = False,
+) -> TensorMap:
+    """
+    Sort the tensor map according to the key values and the blocks for each
+    specified axis in :param axes: according to the label values in the axis.
+
+    Each block is sorted separately, see :py:func:`sort_block` for more information
+
+    Note: This function duplicates metadata on the CPU for the purpose of sorting.
+
+    :param axes: axis of array to argsort
+    :param descending: if false, the order is ascending
+    :return: sorted tensor map
+
+    >>> import numpy as np
+    >>> import metatensor
+    >>> from metatensor import TensorBlock, TensorMap, Labels
+    >>> block_1 = TensorBlock(
+    ...     values=np.arange(9).reshape(3, 3),
+    ...     samples=Labels(
+    ...         ["structures", "centers"], np.array([[0, 3], [0, 1], [0, 2]])
+    ...     ),
+    ...     components=[],
+    ...     properties=Labels(["n", "l"], np.array([[1, 0], [2, 0], [0, 0]])),
+    ... )
+    >>> block_2 = TensorBlock(
+    ...     values=np.arange(3).reshape(1, 3),
+    ...     samples=Labels(["structures", "centers"], np.array([[0, 0]])),
+    ...     components=[],
+    ...     properties=Labels(["n", "l"], np.array([[1, 0], [2, 0], [0, 0]])),
+    ... )
+    >>> tm = TensorMap(
+    ...     keys=Labels(["species"], np.array([[1], [0]])), blocks=[block_2, block_1]
+    ... )
+    >>> metatensor.sort(tm)
+    TensorMap with 2 blocks
+    keys: species
+             0
+             1
+    """
+    blocks: List[TensorBlock] = []
+
+    sorted_idx = _dispatch.argsort_labels_values(tensor.keys.values, reverse=descending)
```

Comment:
Should not sort the keys by default? and allow the user to select if they want to sort it or only sort the blocks